app/my_jss/music/domConnectPlaylistFunc.js

```diff
@@ -0,0 +1,100 @@
+initializePlaylist();
+
+function initializePlaylist(){
+
+    var playlistsParent = document.getElementsByClassName('section-absolute')[0];
+
+    function mapPlaylists(val){
+      return renderPlaylist(val,playlistsParent);
+    }
+
+    playlistsManager.allPlaylists.map(mapPlaylists);
+}
+
+function renderPlaylist(objectToRender,parentObject){
+
+  var smallDiv = document.createElement('div');
+  smallDiv.setAttribute("class","individual-playlist");
+  parentObject.appendChild(smallDiv);
+
+  smallDiv.innerHTML =
+  '<div class="individual-playlist">'+
+      ' <div id="individual-playlist-button">'+
+          '<img src="../core/assets/play-icon.png">'+
+          '<p>Explore Playlist</p>'+
+      ' </div> '+
+  '<img src="'+ objectToRender.imageLarge +'"> '+
+  '<h3>'+objectToRender.title+'</h3>'+
+   '<p> '+objectToRender.description+'</p>'+
+  '</div>';
+
+  var buttonInPlaylist = smallDiv.getElementsByTagName("div")[0];
+
+  buttonInPlaylist.addEventListener('click', function(){
+    renderSpecificPlaylistObjects(objectToRender);
+  });
+}
+
+// render every playlist given from
+function renderSpecificPlaylistObjects(objectToRender) {
+
+  var parentObjectRended = document.getElementById('playlistAbsolut');
+  parentObjectRended.classList.add("playlistDisplayInline");
+
+  var mapSongs = function(val){
+
+    return renderSong(val);
+  };
+
+  var reduceSong = function(memo,val){
+    return memo+val;
+  };
+
+  parentObjectRended.innerHTML =
+  '<div class="playlist-top"><a>CLOSE</a><img src="../core/assets/close-icon.png">'+
+'  </div>'+
+    ' <div class="playlist-main">'+
+    '    <img src="'+ objectToRender.imageSmall+'">' +
+      '  <div>'+
+        '   <p>Playing from: </p>'+
+    '       <h2>'+ objectToRender.title +': The remixes radio</h2>'+
+    '    </div>'+
+    ' </div>'+
+    ' <table class="table_songs">'+
+    '    <tr id="headTable">'+
+      '     <th>SONGS</th>'+
+      '     <th>LENGTH</th>'+
+      '     <th>LISTENED</th>'+
+      '  </tr>'+
+      '  <tr class="tr-second-in-table">'+
+      '<tr>'+
+      objectToRender.songs.map(mapSongs).reduce(reduceSong,"")+
```

Comment:
Will reduce an array of strings into a String, so i can render the songs.

On 26.03.2016 22:07, Cazaciuc Gabriel wrote:
>
> In app/my_jss/music/domConnectPlaylistFunc.js 
> <https://github.com/SOFTVISION-University/cluj-pre-js-2016/pull/110#discussion_r57518794>:
>
> > +    ' <div class="playlist-main">'+
> > +    '    <img src="'+ objectToRender.imageSmall+'">' +
> > +      '  <div>'+
> > +        '   <p>Playing from: </p>'+
> > +    '       <h2>'+ objectToRender.title +': The remixes radio</h2>'+
> > +    '    </div>'+
> > +    ' </div>'+
> > +    ' <table class="table_songs">'+
> > +    '    <tr id="headTable">'+
> > +      '     <th>SONGS</th>'+
> > +      '     <th>LENGTH</th>'+
> > +      '     <th>LISTENED</th>'+
> > +      '  </tr>'+
> > +      '  <tr class="tr-second-in-table">'+
> > +      '<tr>'+
> > +      objectToRender.songs.map(mapSongs).reduce(reduceSong,"")+
>
> not following... what does this map, reduce suppose to do ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/SOFTVISION-University/cluj-pre-js-2016/pull/110/files/50614d9f5b9a26441c9fb534a63ba03bdc1518fb#r57518794>
>